File: ueb1/server/Messages.hs
{-# LANGUAGE DeriveGeneric #-}

module Messages (
  ServerMessage (NewGame, PositionUpdate, ShowMessage),
  ClientMessage (Move)
) where

import Maze
import GHC.Generics

-- Nachrichten die vom Server zum Client gesendet werden

data ServerMessage = NewGame { state :: GameState }
                   | PositionUpdate { pos :: Position }
                   | ShowMessage { text :: String } deriving (Generic)

-- Nachrichten die vom Client zum Server gesendet werden

data ClientMessage = Move { direction :: Direction } deriving (Generic)

File: ueb1/client/src/main/scala/Messages.scala
object Messages {  
  // Nachrichten die vom Server zum Client gesendet werden
  sealed trait ServerMessage

  object Server {
    case class NewGame(state: GameState) extends ServerMessage
    case class PositionUpdate(pos: Position) extends ServerMessage
    case class ShowMessage(text: String) extends ServerMessage  
  }

  // Nachrichten die vom Client zum Server gesendet werden
  sealed trait ClientMessage

  object Client {
    case class Move(direction: Direction) extends ClientMessage
  }
}